src/core/thread/thread_netif.cpp

```diff
@@ -171,6 +172,23 @@ ThreadError ThreadNetif::RouteLookup(const Ip6::Address &source, const Ip6::Addr
     return error;
 }
 
+ThreadError ThreadNetif::TmfFilter(const otMessage *aMessage, const otMessageInfo *aMessageInfo)
+{
+    const Ip6::MessageInfo &messageInfo = *static_cast<const Ip6::MessageInfo *>(aMessageInfo);
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(messageInfo.GetSockAddr().IsRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsRoutingLocator() ||
+                 messageInfo.GetSockAddr().IsLinkLocal() ||
+                 messageInfo.GetSockAddr().IsAnycastRoutingLocator() ||
+                 messageInfo.GetPeerAddr().IsAnycastRoutingLocator(),
```

Comment:
I like the thinking.  However, I still think we need to align the logic better with the specification.

The logic above allows a message to pass if any of the above conditions are true.  For example, the spec says that the source address must be an RLOC or ALOC.  However, the logic above will allow any source address as long as the destination address matches one of the above.